examples/missing_values.py

```diff
@@ -32,41 +33,56 @@
 
 rng = np.random.RandomState(0)
 
-dataset = load_boston()
+dataset = load_diabetes()
 X_full, y_full = dataset.data, dataset.target
 n_samples = X_full.shape[0]
 n_features = X_full.shape[1]
 
+#Create a random matrix to randomly make missing values
+missing_matrix = rng.rand(n_samples, n_features)
+
+# each sample has (1-th)^n_features of probability to have full features
+th = 0.14
+mask = missing_matrix < th
+missing_samples = mask.any(axis=1)
+full_percentage = (n_samples - missing_samples.sum())/float(n_samples)
+print("Percentage of samples with full features: %f" %full_percentage )
+
 # Estimate the score on the entire dataset, with no missing values
+
 estimator = RandomForestRegressor(random_state=0, n_estimators=100)
 score = cross_val_score(estimator, X_full, y_full).mean()
 print("Score with the entire dataset = %.2f" % score)
 
-# Add missing values in 75% of the lines
-missing_rate = 0.75
-n_missing_samples = np.floor(n_samples * missing_rate)
-missing_samples = np.hstack((np.zeros(n_samples - n_missing_samples,
-                                      dtype=np.bool),
-                             np.ones(n_missing_samples,
-                                     dtype=np.bool)))
-rng.shuffle(missing_samples)
-missing_features = rng.randint(0, n_features, n_missing_samples)
 
 # Estimate the score without the lines containing missing values
+
 X_filtered = X_full[~missing_samples, :]
 y_filtered = y_full[~missing_samples]
 estimator = RandomForestRegressor(random_state=0, n_estimators=100)
 score = cross_val_score(estimator, X_filtered, y_filtered).mean()
 print("Score without the samples containing missing values = %.2f" % score)
 
-# Estimate the score after imputation of the missing values
+# Estimate the score after mean imputation of the missing values
+
 X_missing = X_full.copy()
-X_missing[np.where(missing_samples)[0], missing_features] = 0
+X_missing[mask] = np.nan
```

Comment:
Oh I see. Because in original dataset there might be 0 entries, and if I label missing entries as 0, those entries will also be imputed.